test/new-e2e/tests/containers/dump_cluster_state.go

```diff
@@ -286,4 +289,55 @@ func dumpK8sClusterState(ctx context.Context, kubeconfig *clientcmdapi.Config, o
 		fmt.Fprintf(out, "Failed to execute Get command: %v\n", err)
 		return
 	}
+
+	// Get the logs of containers that have restarted
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigFile.Name())
+	if err != nil {
+		fmt.Fprintf(out, "Failed to build Kubernetes config: %v\n", err)
+		return
+	}
+	k8sClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Fprintf(out, "Failed to create Kubernetes client: %v\n", err)
+		return
+	}
+
+	continueToken := "start"
+	for continueToken != "" {
+		if continueToken == "start" {
+			continueToken = ""
+		}
+		pods, err := k8sClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{
+			Continue: continueToken,
```

Comment:
Is this required? I thought that by default it returns everything?

Reply:
If the default limit is “unlimited”, let’s get rid of paging support.